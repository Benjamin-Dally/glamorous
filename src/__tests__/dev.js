/*
 * This file is here to test things that glamor does when
 * NODE_ENV === 'development'
 *
 * Because glamor stores the value in the module level of
 * it's implementation files, we have to do some funny things
 * to make sure that glamor gets the value we want.
 */
import React from 'react'
import {render} from 'enzyme'

const nodeEnv = process.env.NODE_ENV

jest.mock('../constants')

beforeEach(() => {
  process.env.NODE_ENV = 'development'
  jest.resetModules()
})

afterEach(() => {
  process.env.NODE_ENV = nodeEnv
})

test('can be configured to use the displayName in the className', () => {
  // eslint-disable-next-line
  const glamorous = require('../').default
  const MyComp = glamorous(props => <div {...props} />, {
    displayName: 'Hi There',
  })({color: 'red'})
  const wrapper = render(<MyComp />)
  expect(wrapper).toMatchSnapshot()
  expect(wrapper[0].attribs.class).toMatch(/hi-there/)
})
