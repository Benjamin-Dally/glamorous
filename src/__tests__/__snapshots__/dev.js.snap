// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can be configured to use the displayName in the className 1`] = `
.glamor-0,
[data-glamor-0] {
  color: red;
}

<div
  class="glamor-0"
/>
`;
